### ISSUE TITLE
(Mis-)behaviour report: 'End session' overrides previous bluetooth state

### ISSUE THREAD

**Issue by goshuberto** [GrapheneOS/os-issue-tracker#3342]
Hi,

I'm not sure whether the Bluetooth state handling on my Pixel Tablet is intentional or a bug - I'd expect it to be consistent with how Bluetooth state is preserved for reboots (if it was on, it'ill be on again after reboot finishes (in BFU state)).

If I 'End session', BT gets switched off - similar to what I observed ever since using Android.
On my Pixel Tablet, GrapheneOS doesn't re-enable Bluetooth (switched back to admin account - BT stays off), although it was enabled previously (before switching user profile, even in BFU state it was enabled).
Auto-off is set to 'never'.

My previous tablets re-enabled BT afrter profile-switching (before unlock), so I could enter the password via BT-keyboard.  That's the annoying part with my Pixel Tablet running GrapheneOS:  I need to enter the password by touch-screen.
From a security point of view, this might be desired.  But if user decides to use BT-keyboard, I'd prefer to leave it up to users' decision whether to use it for unlocking or not.

Can someone please tell if thos BT-rfkill-override is really intentional?

Thanks in advance!

**Comment by snrkl** [comment]
@goshuberto do you think this is a duplicate of #2667?

**Comment by goshuberto** [comment]
> @goshuberto do you think this is a duplicate of #2667?

Yes, what hurried6158 describes for his Pixel 7 is exactly what I observed very displeasingly with the Pixel Tablet and my BT-keyboard.
I haven't had the opportunity to verify the same behavior on my Pixel 7 yet - I thought it isn't present there.  Thinking twice, it's apparent that it doesn't affect me on the Pixel 7 because for me there's no reason to turn BT on in the admin profile, hence it is always off.

So the only benefit of the current behavior I can see is if somebody forgets to turn BT off before switching to secondary profile.

But to assist the more forgetful people (like me), there's already the auto-BT-off feature, which probably would have turned BT off before user switches to secondary profile - if admin-session wasn't too short...
(but only 1st admin-session would need to be long enough for the auto-off to kick in, since state is persistent across reboots; so 1st reboot results in adminBT-off-state - if 'end session' behavior would (correctly) keep the state, the effect wouldn't be different).

I vote for making it consistent:  'end session' needs to restore previous state like it's already the case for rebooting.

Just checked: On another non-GOS Android 13 device (missing the 'end session' of course) BT-state is kept device-specific, NOT user-profile specific.
It is turned off before/during switching profiles and turned on again **before user unlocks the target profile**.

It looks like 'end session' feature prevents the stock re-enabling behavior.
If this is intentional, it would be nice if there's a switch to allow stock-behavior!

@snrkl  do you close this one as duplicate?  I don't know how to - just getting 'completed' or not 'planned' for closing reasons - which doesn't match, especially the latter hopefully...

**Comment by snrkl** [comment]
@goshuberto just mark it as a duplicate - someone with privileges will have to clean it up.. I don't think regular folks can..

Duplicate of #2667 - can be closed.
